tests/image_test.py

```diff
@@ -52,19 +50,15 @@
 
 class ImageTest(jtu.JaxTestCase):
 
-  @parameterized.named_parameters(jtu.cases_from_list(
-       {"testcase_name": "_shape={}_target={}_method={}_antialias={}".format(
-          jtu.format_shape_dtype_string(image_shape, dtype),
-          jtu.format_shape_dtype_string(target_shape, dtype), method,
-          antialias),
-        "dtype": dtype, "image_shape": image_shape,
-        "target_shape": target_shape,
-        "method": method, "antialias": antialias}
-       for dtype in float_dtypes
-       for target_shape, image_shape in itertools.combinations_with_replacement(
-        [[2, 3, 2, 4], [2, 6, 4, 4], [2, 33, 17, 4], [2, 50, 38, 4]], 2)
-       for method in ["nearest", "bilinear", "lanczos3", "lanczos5", "bicubic"]
-       for antialias in [False, True]))
+  _TF_SHAPES = [[2, 3, 2, 4], [2, 6, 4, 4], [2, 33, 17, 4], [2, 50, 38, 4]]
+
+  @jtu.sample_product(
+    dtype=float_dtypes,
+    target_shape=_TF_SHAPES,
+    image_shape=_TF_SHAPES,
```

Comment:
Just noting: this iterates over the product of shapes rather than combinations, which I think is OK? But it's a different set of test cases than before.

I think it's fine.

Reply:
I think the intent before was to produce the full product! I think it was actually a bug to use combinations_with_replacement.